src/app.ts

```diff
@@ -0,0 +1,28 @@
+// Disable import first to allow dotenv configuration to happen before any imports
+/* eslint-disable import/first */
+import * as dotenv from 'dotenv';
+dotenv.config();
+
+import { log } from './lib/log';
+import { buildServer } from './server';
+
+const APPLICATION_HOST: string = process.env.APPLICATION_HOST as string;
+const APPLICATION_PORT = Number(process.env.APPLICATION_PORT);
```

Comment:
Backend kaatuu jos APPLICATION_PORT ympäristömuuttujaa ei ole määritelty järjestelmätasolla.

`$ npm run start-dev

> blenderi-backend@0.0.1 start-dev
> nodemon

[nodemon] 2.0.19
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): src\**\*
[nodemon] watching extensions: ts,json
[nodemon] starting `ts-node ./src/app.ts`
{
  code: 'ERR_SOCKET_BAD_PORT',
  level: 'error',
  message: 'Error starting server! options.port should be >= 0 and < 65536. Received NaN.',
  stack: 'RangeError [ERR_SOCKET_BAD_PORT]: options.port should be >= 0 and < 65536. Received NaN.\n' +
    '    at new NodeError (node:internal/errors:393:5)\n' +
    '    at validatePort (node:internal/validators:334:11)\n' +
    '    at Server.listen (node:net:1613:5)\n' +
    '    at C:\\Users\\Jaakko\\TampereenUrheilusukeltajatry\\Blenderi-Backend\\node_modules\\fastify\\lib\\server.js:206:14\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at C:\\Users\\Jaakko\\TampereenUrheilusukeltajatry\\Blenderi-Backend\\node_modules\\fastify\\lib\\server.js:205:20\n' +
    '    at processTicksAndRejections (node:internal/process/task_queues:95:5)',
  timestamp: '2022-09-11T09:14:09.675Z'
}
[nodemon] app crashed - waiting for file changes before starting...
`

Reply:
Olis varmaan kiva, jos ympäristömuuttujat määriteltäisiin prosessitasolla, ei _järjestelmätasolla_